### PR TITLE
Issue/cif 759

### PR DESCRIPTION
**Pull request by sandyaggarwal**
## Description

- Fixed Image ratio issue
- Introduced Ordering of Products on the basis of Dialog Box
- Fixed rendering of Price and Currency

## Motivation and Context

Changes are required to fix Image Ratio issue, Ordering of products and Rendering issue

## How Has This Been Tested?

Manual Testing

## Screenshots (if appropriate):
product carousel

## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
- [ x] I have added tests to cover my changes and the overall coverage did not decrease.
- [ x] All unit tests pass on CircleCi.
- [ x] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by herzog31** [comment]
@sandyaggarwal please apply the correct copyright headers in HTL as introduced in https://github.com/adobe/aem-core-cif-components/pull/67.

**Comment by cjelger** [comment]
It would be very useful to add a unit test for the sorting, so we know it does what we expect and one cannot break it without breaking the test.

File: bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductCarouselImplTest.java [hunk 1]
```diff
@@ -99,4 +100,14 @@ public void getProducts() {
...
+    @Test
+    public void testProductListOrder() {
+
+        List<ProductListItem> productList = this.slingModel.getProducts();
+        final List<String> productSkus = Arrays.asList(this.productSkuList);
+
+        Collections.sort(productList, Comparator.comparing(item -> productSkus.indexOf(item.getSKU())));
+        Assert.assertTrue(productSkus.equals(productList.stream().map(item -> item.getSKU()).collect(Collectors.toList())));
+    }
+
```
Comment (sandyaggarwal) [inline bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductCarouselImplTest.java]: @cjelger test added for product sorting

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductCarouselImpl.java [hunk 2]
```diff
@@ -82,8 +81,7 @@ private void initModel() {
                 resource.getPath());
         }
         this.productList = this.fetchProductFromGraphql(magentoGraphqlClient, productSkus);
-        // Sort Product List
-        Collections.sort(productList, new ProductSkuComparator(getSortOrder(productSkus)));
+        Collections.sort(this.productList, Comparator.comparing(item -> productSkus.indexOf(item.getSku())));
```
Comment (sandyaggarwal) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductCarouselImpl.java]: @mhaack This works fine

**Comment by mhaack** [comment]
@sandyaggarwal see @cjelger 's comment, I had a quick look at least `product__card--image` is missing in the readme.md Please double check the other css as well.

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/README.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ BLOCK productcarousel
 
 BLOCK product
     ELEMENT product__card
+    ELEMENT product__card--image
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/README.md]: This is a `MODIFIER`, not an `ELEMENT`

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html [hunk 4]
```diff
@@ -14,9 +14,13 @@
 <template data-sly-template.productcard="${@ product}">
     <div role="displaycard" class="card product__card">
         <a href="${product.URL}">
-            <img class="product__image" src="${product.imageURL}" data-gallery-role="currentimage" />
-            <div>${product.Title}</div>
-            <div class="product__price">${product.Currency} ${product.Price}</div>
+            <div class="product__card--image">
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html]: I don't think this is valid here, the `--image` suffix is a BEM modifier that should define a "behavior" or "state", not a child element.

**Comment by cjelger** [comment]
@dplaton and @herzog31 can you maybe help here with the BEM stuff? I'm also not sure if it's the riht place for the custom `css`, that is, our custom changes vs. css extracted from venia components. Thank you.

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html (hunk 4 above)
Comment (dplaton) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/productcard.html]: In this case, it should be `product__card__image`

**Comment by dplaton** [comment]
@sandyaggarwal Custom CSS should go into a separate CSS file (even if it overwrites the classes)

**Comment by sandyaggarwal** [comment]
@dplaton @cjelger This component  is not using any css from Venia and has all the custom css in file product-carousel.css so this is  altogether a  separate file with distinct css classes.

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/README.md [hunk 5]
```diff
@@ -48,6 +48,7 @@ BLOCK productcarousel
 
 BLOCK product
     ELEMENT product__card
+    ELEMENT product__card__image
     ELEMENT product__image
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/productcarousel/v1/productcarousel/README.md]: I think you removed `product__image` right?

**Pull request merged**